src/bin/pager/src/main.rs

```diff
@@ -72,3 +95,80 @@ fn main() {
     .detach();
     twizzler_async::run(std::future::pending::<()>());
 }
+
+static mut RAND_STATE: u32 = 0;
+pub fn quick_random() -> u32 {
+    let state = unsafe { RAND_STATE };
+    let newstate = state.wrapping_mul(69069).wrapping_add(5);
+    unsafe {
+        RAND_STATE = newstate;
+    }
+    newstate >> 16
+}
+
+struct Tester<'a> {
+    kv: KeyValueStore<'a>,
+    truth: BTreeMap<Key, Foo>,
+}
+
+#[allow(dead_code)]
+impl<'a> Tester<'a> {
+    fn test(&mut self) {
+        const TEST_ITERS: u32 = 100000;
+        for i in 0..TEST_ITERS {
+            // Every once in a while, validate some things.
+            if i % 2000 == 0 {
+                self.validate_has_all();
+            }
+            let x = i % (10001 + i / 1000);
+            let k = Key::new(ObjID::new(0), x, store::KeyKind::ObjectInfo);
+            let _ = self.get(k);
+            let num = quick_random() % 3;
+            if num == 0 || num == 2 {
+                let _ = self.put(k, Foo { x: x });
+            } else if num == 1 {
+                let _ = self.del(k);
+            }
+        }
+    }
+
+    fn validate_has_all(&self) {
+        for (key, val) in self.truth.iter() {
+            let res: Foo = self.kv.get(*key).unwrap();
+            assert_eq!(res, *val);
+        }
+    }
+
+    fn get(&self, key: Key) -> Result<Foo, ErrorCode> {
+        let r = self.kv.get(key);
+        if r.is_ok() {
+            assert!(self.truth.contains_key(&key));
+            let t = self.truth.get(&key).unwrap();
+            assert_eq!(t, r.as_ref().unwrap());
+        } else {
+            assert!(!self.truth.contains_key(&key));
+        }
+        r
```

Comment:
Can we make this "res" across all these functions so they a) match and b) the name isn't a single letter?

Reply:
Done!